pruner/src/statement_pruner.cpp

```diff
@@ -21,6 +21,19 @@ const IR::Node *Pruner::preorder(IR::BlockStatement *s) {
     return s;
 }
 
+const IR::Node *Pruner::preorder(IR::IfStatement *s) {
+    // If ifTrue is only one statement, try to replace
+    // it with the two branches
+    if (!s->ifTrue->is<IR::BlockStatement>()) {
```

Comment:
I think we can do more here, right? Remove one of the branches, or remove both. Why does this lead to a bug?

Reply:
We can surely do that, the bug is only triggered when the if block is empty, so something like this :- 

```c
    if(condition 1){

       }

   else {
          statement_1;
          .
          .
          statement_n;
    }
```

Reply:
I think this may be an assertion in the if statement. If that happens. The true part of the statement can not be nullptr. So we just can prune the else part, convert the entire if statement into a simple block statement, or replace the true part with the else part (and negate the condition)

Reply:
This looks incomplete to me. Shouldn't we either delete both branches, or one or the other? Currently this always deletes the else statement.  If you delete the then part of the statement you just need to swap in the else part. If there is no else part, you can delete the entire statement. So you need a random dice roll here.